bin/compute-classpath.sh

```diff
@@ -121,10 +124,15 @@ datanucleus_jars="$(find "$datanucleus_dir" 2>/dev/null | grep "datanucleus-.*\\
 datanucleus_jars="$(echo "$datanucleus_jars" | tr "\n" : | sed s/:$//g)"
 
 if [ -n "$datanucleus_jars" ]; then
-  hive_files=$("$JAR_CMD" -tf "$ASSEMBLY_JAR" org/apache/hadoop/hive/ql/exec 2>/dev/null)
-  if [ -n "$hive_files" ]; then
-    echo "Spark assembly has been built with Hive, including Datanucleus jars on classpath" 1>&2
-    CLASSPATH="$CLASSPATH:$datanucleus_jars"
+  if [ $(command -v "$JAR_CMD") ] ; then
+    hive_files=$("$JAR_CMD" -tf "$ASSEMBLY_JAR" org/apache/hadoop/hive/ql/exec 2>/dev/null)
+    if [ -n "$hive_files" ]; then
+      echo "Spark assembly has been built with Hive, including Datanucleus jars on classpath" 1>&2
+      CLASSPATH="$CLASSPATH:$datanucleus_jars"
+    fi
+  else
+    echo "No jar command available; unable to check if assembly was built with Hive."
```

Comment:
Hmm... not sure about this one. The thrift server probably won't run, in many cases, if the datanucleus jars are not added to its classpath. So won't apps that use HiveContext with a local metastore, for example. Perhaps this should be an error instead? Of maybe try to be smart and look at `$JAVA_HOME/../bin/jar` too?

(BTW, shameless plug: SPARK-4924 fixes this. But it's not in 1.3.)